Configuration/MasteryConfig.cs

```diff
@@ -33,5 +33,12 @@ public static void Initialize()
         WeaponMasterySystem.EffectivenessSubSystemEnabled = _configFile.Bind("Mastery", "Enable Effectiveness Subsystem", true, "Enables the Effectiveness mastery subsystem, which lets you reset your mastery to gain a multiplier to the effects of the matching mastery.").Value;
         WeaponMasterySystem.GrowthPerEffectiveness = _configFile.Bind("Mastery", "Growth per effectiveness", 1.0, "Used to determine how mastery growth is affected at higher effectiveness levels. When enabled, more effectiveness will mean that it is slower to gain mastery (at 1, 200% effectiveness gives a growth rate of 50%). A smaller number here will reduce the growth rate faster. Use a value of 0 to disable. Negative values are converted to positive.").Value;
         WeaponMasterySystem.MaxEffectiveness = _configFile.Bind("Mastery", "Maximum Effectiveness", 5.0, "The maximum mastery effectiveness where 1 is 100%.").Value;
+        WeaponMasterySystem.OffensiveStatIncreaseFactor = _configFile.Bind("Mastery", "Offensive Stat Increase Factor", 10, "The percentage increase in offensive stats for every 100% mastery").Value;
+
+        WeaponMasterySystem.AdditionalSpellPowerPer100PercentMastery = _configFile.Bind("Spell Mastery", "Additional Spell Power per 100% Mastery", 10, "The additional amount of spell power gained per 100% mastery.").Value;
+        WeaponMasterySystem.AdditionalSpellCriticalStrikeChancePer100PercentMastery = _configFile.Bind("Spell Mastery", "Additional Spell Critical Strike Chance per 100% Mastery", 5, "The additional amount of spell critical strike chance gained per 100% mastery.").Value;
+        WeaponMasterySystem.AdditionalSpellCriticalStrikeDamagePer100PercentMastery = _configFile.Bind("Spell Mastery", "Additional Spell Critical Strike Damage per 100% Mastery", 25, "The additional amount of spell critical strike damage gained per 100% mastery.").Value;
+        WeaponMasterySystem.AdditionalSpellLifeLeechPer100PercentMastery = _configFile.Bind("Spell Mastery", "Additional Spell Life Leech per 100% Mastery", 3, "The additional amount of spell life leech gained per 100% mastery.").Value;
+        WeaponMasterySystem.AdditionalSpellCooldownRecoveryRatePer100PercentMastery = _configFile.Bind("Spell Mastery", "Additional Spell Cooldown Recovery Rate per 100% Mastery", 10, "The additional amount of spell cooldown recovery rate gained per 100% mastery.").Value;
```

Comment:
If you want to go down the route of this configuration, then I would suggest it would be better to use the existing `Database.MasteryStatConfig` so we don't have to clutter the config with large lists of basically the same config.

You can set the default rates of improvement for each weapon type in the `WeaponMasterySystem.DefaultMasteryConfig`, which then gets written out to `Data\weaponMasteryConfig.json` and is user editable.

This default config is essentially using the following config:
```
{ MasteryType.Spell, new List<StatConfig>()
                {
                    new(UnitStatType.SpellPower, 0,  0.1f ),
                    new(UnitStatType.SpellCriticalStrikeChance, 0,  0.05f ),
                    new(UnitStatType.SpellCriticalStrikeDamage, 0,  0.25f ),
                    new(UnitStatType.SpellLifeLeech, 0,  0.03f ),
                    new(UnitStatType.SpellCooldownRecoveryRate, 0,  0.1f )
                } }
```

Reply:
I think this is a good approach because, as far as I can see, we don't have a weapon or something similar to buff regarding spell power. I've modified my solution to use the masteryStatConfig.